tests/data/test_huggingface_dataset_and_source.py

```diff
@@ -17,6 +19,30 @@
 from mlflow.types.utils import _infer_schema
 
 
+@pytest.fixture(scope="module", autouse=True)
+def mock_datasets_load_dataset():
+    """
+    `datasets.load_dataset` is flaky and sometimes fails with a network error.
+    This fixture retries the call up to 5 times with exponential backoff.
+    """
+
+    original = datasets.load_dataset
+
+    def load_dataset(*args, **kwargs):
+        for i in range(5):
+            try:
+                return original(*args, **kwargs)
+            except Exception:
+                if i < 4:
+                    time.sleep(2**i)
+                    continue
+                raise
+
+    with mock.patch("datasets.load_dataset", wraps=load_dataset) as mock_load_dataset:
+        yield
+        mock_load_dataset.assert_called()
```

Comment:
Do we need this? What if load_dataset doesn't exist in one test?

Reply:
That's fine because `session` is `module`. It must be called in this test file.